Pony/source/Socket.cpp

```diff
@@ -79,58 +84,56 @@ bool Socket::Open(unsigned short port)
 
 #if defined(_WIN32)
     DWORD nonBlocking = 1;
-    if (ioctlsocket( socket, FIONBIO, &nonBlocking) != 0) {
-	printf( "failed to set non-blocking socket\n" );
-	Close();
-	return false;
-    }
+    if (ioctlsocket(m_socket, FIONBIO, &nonBlocking) != 0) {
 #else
     int nonBlocking = 1;
     if (fcntl(m_socket, F_SETFL, O_NONBLOCK, nonBlocking) == -1) {
+#endif
         Close();
         return false;
     }
-#endif
+
     return true;
 }
 
-int Socket::Send(const Address & dst, const void * data, unsigned size)
+signed Socket::Send(const Address& dst, const void* data, unsigned size)
 {
-    if ((m_socket == 0) || ( ! data) || (size <=0))
+    if ((m_socket == 0) || (!data) || (size <= 0))
         return -1;
 
     sockaddr_in address;
     address.sin_family = AF_INET;
     address.sin_addr.s_addr = htonl(dst.GetAddress());
-    address.sin_port        = htons((unsigned short) dst.GetPort());
+    address.sin_port = htons(dst.GetPort());
 
-    int sended_bytes = sendto(m_socket, (const char*)data, size, 0, (sockaddr*)&address, sizeof(sockaddr_in));
-    if (size - sended_bytes)
+    unsigned sended = sendto(m_socket, static_cast<const char*>(data), size, 0, (sockaddr *) &address, sizeof(sockaddr_in));
+    if (size - sended)
         return 0;
 
-    return sended_bytes;
+    return sended;
 }
 
-int Socket::Recv(Address & sender, void * data, unsigned size)
+signed Socket::Recv(Address& sender, void* data, unsigned size)
 {
-    if ((m_socket == 0) || ( ! data) || (size <=0))
+    if ((m_socket == 0) || (!data) || (size <= 0))
```

Comment:
Size is unsigned. It cannot be less than zero.

Reply:
ok